backend/tracim_backend/app_models/applications.py

```diff
@@ -1,80 +1,59 @@
 # coding=utf-8
 import typing
 
-from tracim_backend.app_models.contents import ContentType
-from tracim_backend.models.roles import WorkspaceRoles
-
 if typing.TYPE_CHECKING:
-    from tracim_backend.config import CFG  # noqa: F401
-    from tracim_backend.app_models.contents import ContentStatus  # noqa: F401
+    from tracim_backend.config import CFG
+    from tracim_backend.lib.utils.app import TracimApplication
+    from tracim_backend.lib.utils.app import TracimContentType  # noqa F401
 
 
-class Application(object):
+class TracimApplicationInContext(object):
     """
     Application class with data needed for frontend
     """
 
-    def __init__(
-        self,
-        label: str,
-        slug: str,
-        fa_icon: str,
-        is_active: bool,
-        config: typing.Dict[str, str],
-        main_route: str,
-        app_config: "CFG",
-    ) -> None:
-        """
-        @param label: public label of application
-        @param slug: identifier of application
-        @param fa_icon: font awesome icon class
-        @param is_active: True if application enable, False if inactive
-        @param config: a dict with eventual application config
-        @param main_route: the route of the frontend "home" screen of
-        the application. For exemple, if you have an application
-        called "agenda", the main route will be something
-        like /workspace/{wid}/agenda.
-        """
-        self.label = label
-        self.slug = slug
-        self.fa_icon = fa_icon
-        self.hexcolor = self._get_hexcolor_or_default(slug, app_config)
-        self.is_active = is_active
-        self.config = config
-        self.main_route = main_route
-        self.content_types = []
+    def __init__(self, app_config: "CFG", app: "TracimApplication") -> None:
+        self.app = app
+        self.app_config = app_config
```

Comment:
About `app_config` naming: it looks like a config specific for this app. But it is a global Tracim config right ?

Reply:
Yes. 

This name is widely use to distinct "config" and "settings" object which exist at some point of code because of usage of some lib (pyramid as a configurator object, caldav app use a "config" dict, settings is the current setting dict used by pyramid, etc...)

Until recently there was not naming issue because application doesn't really exist.

Reply:
ok